test/functional/terminal/job_spec.lua

```diff
@@ -31,9 +30,6 @@ describe('job handler', function()
         elseif 'stderr' ==# a:event
           put =join(a:data,"\n")
         elseif 'exit' ==# a:event
-          " Length of received data.
-          call setline('.', 'bytes: '.len(join(g:job_stdout[0:-2],"\n")))
```

Comment:
@jamessan There might not be a bug here after all. Instead this might be the result of the behavior described here: https://github.com/neovim/neovim/issues/4897#issuecomment-224754553

Because the job handlers here are blindly joining `a:data` without regards to whether `a:data` continues a previous partial result.